SYCL/USM/dep_events.cpp

```diff
@@ -0,0 +1,47 @@
+//==---- dep_events.cpp - USM dependency test ------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple  %s -o %t1.out
+// RUN: %HOST_RUN_PLACEHOLDER %t1.out
+// RUN: %CPU_RUN_PLACEHOLDER %t1.out
+// RUN: %GPU_RUN_PLACEHOLDER %t1.out
+// RUN: %ACC_RUN_PLACEHOLDER %t1.out
+
+#include <CL/sycl.hpp>
+
+using namespace sycl;
+
+int main() {
+  queue q;
+
+  int *x = malloc_shared<int>(1, q);
+  int *y = malloc_shared<int>(1, q);
+  int *z = malloc_shared<int>(1, q);
+
+  event eMemset1 = q.memset(x, 0, sizeof(int), event{});              // x = 0
+  event eMemset2 = q.memset(y, 0, sizeof(int), std::vector<event>{}); // y = 0
+  event eFill = q.fill(x, 1, 1, {eMemset1, eMemset2});                // x = 1
+  event eMemcpy = q.memcpy(y, x, sizeof(int), eFill);                 // y = 1
+  event Event = q.copy(y, z, 1, eMemcpy);                             // z = 1
+
+  // TODO cuda_piextUSMEnqueuePrefetch does not yet handle flags
```

Comment:
Can you add **// UNSUPPROTED: cuda** instead of the check in the code which is hard to find.